lib/src/test/java/com/auth0/jwt/JWTVerifierTest.java

```diff
@@ -494,6 +494,33 @@ public void shouldValidateIssuedAtWithLeeway() throws Exception {
         assertThat(jwt, is(notNullValue()));
     }
 
+	// Issued At with future date
+	@Test (expected = InvalidClaimException.class)
+	public void shouldThrowOnFutureIssuedAt() throws Exception {
+		Clock clock = mock(Clock.class);
+		when(clock.getToday()).thenReturn(new Date(DATE_TOKEN_MS_VALUE - 1000));
+
+		String token = "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpYXQiOjE0Nzc1OTJ9.CWq-6pUXl1bFg81vqOUZbZrheO2kUBd2Xr3FUZmvudE";
+		JWTVerifier.BaseVerification verification = (JWTVerifier.BaseVerification) JWTVerifier.init(Algorithm.HMAC256("secret"));
+
+		DecodedJWT jwt = verification.build(clock).verify(token);
+		assertThat(jwt, is(notNullValue()));
+	}
+
+	// Issued At with future date and ignore flag
+	@Test
+	public void shouldNotVerifyIATOnIgnoreIssuedAt() throws Exception {
+		Clock clock = mock(Clock.class);
+		when(clock.getToday()).thenReturn(new Date(DATE_TOKEN_MS_VALUE - 1000));
+
+		String token = "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpYXQiOjE0Nzc1OTJ9.CWq-6pUXl1bFg81vqOUZbZrheO2kUBd2Xr3FUZmvudE";
+		JWTVerifier.BaseVerification verification = (JWTVerifier.BaseVerification) JWTVerifier.init(Algorithm.HMAC256("secret"));
+		verification.ignoreIssuedAt();
+
+		DecodedJWT jwt = verification.build(clock).verify(token);
+		assertThat(jwt, is(notNullValue()));
+	}
+
```

Comment:
Please add another test that calls `acceptsIssuedAt(..)` passing a value and then `ignoreIssuedAt()`. This test would expect no exceptions. You'll see how this test would fail unless you put the claim check in the right place (see my review's general comment)

Reply:
Understand and agree with you on acceptsIssuedAt()

And since the flag is available at builder level, try to achieve the job at the claims map level rather at verification level later in parent class. Please see the updated code. Thank you.